### PR TITLE
Typed docs for sharding and singleton

### PR DESCRIPTION
**Pull request by chbatey**
Leaving distributed data until the new message adapters are merged

Refs #24166

### CODE REVIEW

File: akka-cluster-sharding-typed/src/test/java/jdoc/akka/cluster/sharding/typed/ShardingCompileOnlyTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import akka.cluster.sharding.typed.*;
+import akka.cluster.typed.ClusterSingleton;
+import akka.cluster.typed.ClusterSingletonSettings;
+
+public class ShardingCompileOnlyTest {
+
+  //#counter
+  interface CounterCommand {}
+  public static class Increment implements CounterCommand { }
+  public static class GoodByeCounter implements CounterCommand { }
```
Comment (johanandren) [inline akka-cluster-sharding-typed/src/test/java/jdoc/akka/cluster/sharding/typed/ShardingCompileOnlyTest.java]: Not used?

**Review by johanandren — APPROVED** [review]
Nice!

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by chbatey** [comment]
PLS BUILD

**Comment by johanandren** [comment]
PLS BUILD

**Pull request merged**